core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -218,4 +233,32 @@ static class Fixture extends SqlTypeFixture {
     assertEquals(28, dataType.getPrecision());
     assertEquals(10, dataType.getScale());
   }
+
+  @Test void testCustomMaxTimestampPrecisionTimeFunctionReturnTypeInference() {
```

Comment:
would be helpful to leave a comment with the jira case. there are lots of examples with `Test case for ...` you can use

Reply:
Done